### ISSUE TITLE
Add an embedded test crate to `primitives`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3141]
We need to do a no-std test for `primitives` so we catch bugs like that fixed in #3140.

**Comment by Kixunil** [comment]
Note that this needs to be no-alloc test, not just no-std.

**Comment by tcharding** [comment]
Probably both, one with an allocator and one without, right? I thought we did that somewhere else but I can't find it right now.

**Comment by Kixunil** [comment]
No-alloc test implies `no_std` to catch the kind of bugs like #3140 so one is sufficient.
